src/wallet/rpcwallet.cpp

```diff
@@ -4551,9 +4551,9 @@ UniValue z_sendmany(const UniValue& params, bool fHelp)
             }
         }
 
-        if (!addrStrings.insert(addrStr).second) {
-            throw JSONRPCError(RPC_INVALID_PARAMETER, string("Invalid parameter, duplicated address: ") + addrStr);
-        };
+        if (!recipientAddrs.insert(addr.value()).second) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, string("Invalid parameter, duplicated recipient from address: ") + addrStr);
+        }
```

Comment:
This means we only detect duplicates between selected receivers; if two UAs have the same Orchard receiver but different Sapling receivers, then the UAs can be sent to in the same transaction right up until NU5 activation, and then they will start failing. We _do_ catch all potential duplicates between receivers that end up in actual transactions though, so I think this is fine.

Reply:
I had originally written this to catch all duplicates, not just the preferred receiver but your point makes sense about receivers that end up in actual transactions.